docs/how-to/include_files.rst

```diff
@@ -0,0 +1,178 @@
+.. _how_to_include_files_and_resources:
+
+******************************************
+Including local files and remote resources
+******************************************
+
+Craft-parts provides the built-in :ref:`dump plugin <craft_parts_dump_plugin>`
+for all kinds of projects that need to include local files and remote resources 
+as is.
+This plugin using the :ref:`source <source>` property in the part to download,
+unpack, and copy files and directories from the given source to the build
+environment, then do some organizing if needed, and include them in the final
+payload.
+
+If you don't need to copy these files, consider using the ``nil`` plugin.
+If you need to download and build them from source, consider using plugins for
+the corresponding languages, such as the
+:ref:`python plugin <craft_parts_python_plugin>` and the
+:ref:`rust plugin <craft_parts_rust_plugin>`.
+
+
+The typical use cases for the ``dump`` plugin:
+
+* Include static files in the final payload, such as scripts, configurations,
+  services, documentation, media files, or other resources that are not
+  generated by the build process.
+
+* Include pre-built third-party packages, libraries, binaries, or other
+  artifacts that are not available in the system's package manager or in the
+  project's source code.
+
+The supported source types please refer to :ref:`source_type`.
+
+
+Example: To include a local directory and move files to the correct locations
+-----------------------------------------------------------------------------
+
+Given the following project structure:
+
+.. code-block:: text
+
+  .
+  └── misc
+      ├── fonts
+      │   └── good.otf
+      └── services
+          ├── README
+          └── hello.service
+    
+    
+The following dump part can be used to include the ``hello.service`` and
+``good.otf`` files from the ``misc`` directory, then move them to the correct
+locations, keeping only ``/usr`` in the final payload:
+
+.. code-block:: yaml
+
+  parts:
+    my-part:
+      plugin: dump
+      source: ./misc
+      source-type: local
+      organize:
+        'services/*.service': usr/lib/systemd/system/
+        'fonts/*': usr/share/fonts/
+      stage:
+        - usr/
+
+The resulting payload will look like this:
+
+.. code-block:: text
+
+  .
+  └── usr
+      ├── lib
+      │   └── systemd
+      │       └── system
+      │           └── hello.service
+      └── share
+          └── fonts
+              └── good.otf
+
+
+Example: To include a remote third-party deb package from a URL
+---------------------------------------------------------------
+
+The following dump part downloads a ``busybox-static`` deb package from a
+remote URL and includes it in the final payload:
```

Comment:
It might be worth noting that this doesn't bring in any dependencies.

Reply:
I thought about it, but I'm not sure if I should use a more complex example.